docs/extend/modular-resources/examples/custom-arm.md

```diff
@@ -0,0 +1,258 @@
+
+A configured *module* can make one or more *modular resources* available for configuration.
+
+{{% /alert %}}
+
+## Get your arm's kinematics file
+
+The way arms move across space is more complicated than Viam's other [components](/components/).
+Because of this, the arm, unlike other components, needs a [kinematic configuration file](/internals/kinematic-chain-config/) describing the geometry of the robot arm to work with the RDK's built-in [Motion Service](/services/motion/), on top of the client-inaccessible and mostly static [Frame System Service](/services/frame-system/).
+
+**Find a pre-built kinematics file:**
+
+- The RDK's `viam-server` will work with <file>URDF</file> (United Robot Description Format) kinematics files, which are currently the standard for ROS drivers.
+Many <file>URDF</file> "robot descriptions" for industrial robot arm models can be found on [GitHub](https://github.com/) that could work with the arm on Viam.
+
+**Create your own kinematics file:**
+
+- Follow the instructions on [Configure Complex Kinematic Chains](/internals/kinematic-chain-config/) to write a file detailing the geometry of your arm.
+  - Use the [Spatial Vector Algebra (SVA)](/internals/kinematic-chain-config/#kinematic-parameters) kinematic parameter type.
+  - Define the parameters in a </file>.json</file> file.
+  - Following the Frame System's guide to [Configure a Reference Frame](/services/frame-system/frame-config/) when working out the relative [orientation](/internals/orientation-vector/) of the `"links"` on your arm.
+- View examples of the SVA and JSON format in Viam's built-in arm drivers on [GitHub](https://github.com/viamrobotics/rdk/blob/main/components/arm).
+
+Create a new directory.
+Give it the name your want you call your custom {{< glossary_tooltip term_id="model" text="model" >}} of arm.
+Save this file as <file>your-model-name.json</file> inside of this directory.
+While completing the following step, make sure to save any new files that you generate inside of this directory.
+
+## Create a custom arm model as a modular resource
+
+To create a custom arm model, code a module in Python with the module support libraries provided by [Viam's SDKs](/program/apis/):
+
+1. [Code a new resource model](#code-a-new-resource-model) implementing all methods the Viam RDK requires in the API definition of its built-in subtype (ex. `rdk:component:arm`).
+Import your custom model and API into the main program and register the new resource model with your chosen SDK.
+
+2. [Code a main program](#code-a-main-entry-point-program) that starts the module after adding your desired resources from the registry.
+This main program is the "entry point" to your module.
+
+3. [Compile or package](#compile-the-module-into-an-executable) the module into a single executable that can receive a socket argument from Viam, open the socket, and start the module at the entry point.
```

Comment:
Similarly to above, I think putting the info for a given step in one relevant place is better here. I'd recommend putting the contents of "1. ...." into the "Code a new resource model" section, perhaps with some edits to blend it with the contents there, and same for each of the other steps. That way, people understand what they're doing when they're actually doing it, and don't have to pop up and down the page for context.

I can see the value of what you’re doing here providing some sort of high level outline. Maybe just shorten all these up a bit and move most of the content down as relevant.